File: 백준/Silver/1940. 주몽/주몽.java
import java.util.*;
import java.io.*;
public class Main {
    public static void main(String [] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n= Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());
        int target = Integer.parseInt(st.nextToken());

        long [] array = new long [n];

        st = new StringTokenizer(br.readLine());
        for(int i=0; i<n; i++){
            array[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(array);

        int count=0;
        int start=0;
        int end=n-1;

        long sum = array[start] + array[end];

        while (start<end){
            if(sum==target){
                count+=1;
                sum-=array[start];
                start+=1;
                sum+=array[start];
            }
            else if (sum<target){
                sum-=array[start];
                start+=1;
                sum+=array[start];
            }
            else if(sum > target){
                sum-=array[end];
                end-=1;
                sum+=array[end];
            }
        }
        System.out.println(count);

    }
}

File: 백준/Silver/1940. 주몽/주몽.py
# 주몽의 명령(실버4)
n = int(input())
target = int(input())
ing = list(map(int, input().split()))
ing.sort()
start=0
end=n-1
total = ing[start]+ing[end]
answer=0
while start < end:

    if ing[start]+ing[end] == target:
        end-=1
        start+=1
        answer+=1
    elif ing[start]+ing[end] < target:
        start+=1
    else:
        end-=1
print(answer)